Layout Improvements

Dumeng opened the pull request:
A batch of layout and UI improvements:
- Add a tooltip indictor for Volume bar
- Add line number for song list
- Layout cleanup
- Add platform label back for the search results
- Improve responsive design

listen1 commented on src/components/TrackRow.vue:
```diff
@@ -1,4 +1,7 @@
 <template>
+  <div v-if="index !== undefined" class="flex-none w-8 text-cneter">
```
cneter spell error

listen1 commented:
Thanks for so much work delicated to project. By the way, one pr for one feature is recommended. Easier to merge and modify.

- Add a tooltip indictor for Volume bar (调节音量时显示百分比 #407)

Good to go.

- Add line number for song list (feat:建议给歌单目录歌曲前边加上编号 #698)

Suggstion: Keep same align style for now playing playlist and common playlist.

- Use grid to replace flex layout for playlist list view

Good to go. More elegant way to responsive design👍

- Add default placeholder playlist cover

Question: I notice inline background image uses inline style base64. Will that be performance issue for many images?

- Update player controller bar to flex layout ([建议] 优化下响应式布局，例如将页面缩小后的布局 #643)

This optimaze is necessary. And it's also complicated for design especially in really narrow width screen. I think the layout in narrow screen can be better designed. Or we can limit a min width.

Bug: adjust width of screen, play, next, prev button is not clickable.

- Minor optimize player main info component (is:issue is:open 播放进度条、声音拖动条 点击范围太小，经常点击不到，感谢作者！ #745)

I'm not sure increase playerbar height is for this requirement. It seems too high in my test environment (compare to height of sidebar entries). 

- Add platform label back for the search results

Bug: Seems not working.

- Add Toggle Button component for settings

Necessary improvement. I think underline style seems a little strange because seldom design use that besides tab bar.  

Style suggestion: change underline style to radio button or use iOS style (different background for highlight)

Style suggestion: change checkbox to button group is harder for users to understand. I recommend to use checkbox and we can change icons to more common checkbox icon as they used in netease/qq application.

- Add blur effect to lyrics (歌词在滚动的时候的渐淡消失的效果 #617)

Blur area seems too large. Maybe we can refer to netease/qq effect to adjust parameter

- Refactor code structure of the settings page

Good to go

- Make the settings page responsive

Good to go

- Dynamically import static assets

Good to go

BTW: I'm not good at design, so these design suggestions are not professional. I think what I can do is find out some style many music player applications are using and try to follow them. It's really hard for me to judge a pretty new style is good for users to understand and use.

Dumeng commented:
> Suggstion: Keep same align style for now playing playlist and common playlist.

Fixed

> Bug: adjust width of screen, play, next, prev button is not clickable.

Fixed

> Question: I notice inline background image uses inline style base64. Will that be performance issue for many images?
Refer to (Vite docs for static asset)[https://vitejs.dev/config/#build-assetsinlinelimit]

This may makes final html structure a little messy, but I didn't see performance issue for now.

> Style suggestion: change underline style to radio button or use iOS style (different background for highlight)

Updated

> I'm not sure increase playerbar height is for this requirement. It seems too high in my test environment (compare to height of sidebar entries).

Looks fine for me. Do you have any screenshot can share?

> Style suggestion: change checkbox to button group is harder for users to understand. I recommend to use checkbox and we can change icons to more common checkbox icon as they used in netease/qq application.

I attempt makes the style being more unified. I suggest we can use a standalone toggle button style from android design.
![image](https://user-images.githubusercontent.com/5249513/147726230-4b1b3057-e2df-458d-b23e-05846693fc82.png)

> Blur area seems too large. Maybe we can refer to netease/qq effect to adjust parameter

The `-webkit-mask-image` feature is still experimental. Actually, I blur the lyric line by line. What means 'blur area too large'?

mikelxk commented:
<img width="1691" alt="image" src="https://user-images.githubusercontent.com/32220948/147732053-f4c51ea1-bcef-4452-9c74-25158c2b96dc.png">
It seem that the setting page will break if I set zoom level to 150%.

listen1 commented:
## About blur effect explain

Image 1. Blur effect from netease(mac) music player
<img width="229" alt="截屏2021-12-30 下午10 53 42" src="https://user-images.githubusercontent.com/18599108/147762509-c5aed5fd-c5ff-4b23-9358-b21addda3c86.png">

Image 2. Blur effect from PR
<img width="228" alt="截屏2021-12-30 下午10 56 12" src="https://user-images.githubusercontent.com/18599108/147762707-4c4aa62b-a3f6-4f78-b185-0797a8969e2c.png">

There're some differences.
1. Blur area. 
netease only blur top/bottom small area
2. Blur effect 
netease only have very strong blur effect on top/bottom

mikelxk commented:
Setting Layout still look broken on 150% zoom level for me.
<img width="1919" alt="image" src="https://user-images.githubusercontent.com/32220948/147789712-4643f803-e7db-4836-93b6-58b9d92bd5be.png">
Also, when I enable comment,
<img width="1919" alt="image" src="https://user-images.githubusercontent.com/32220948/147789755-b8046b50-0025-496b-b493-d160cff42f5e.png">
The page is also broken.
I've also attached apple music dealing of lyric blur, as I see, the best implementation so far.
![image](https://user-images.githubusercontent.com/32220948/147789851-1ee1c43d-77d6-4b17-876d-2265c5d8c0cb.png)

Dumeng commented:
> Setting Layout still look broken on 150% zoom level for me.
This is expected. You screenshot is looks like an ultra-wide window view, so the flex flow is extend horizontal. Refer to this [link](https://tailwindcss.com/docs/responsive-design)

mikelxk commented on src/components/TrackRow.vue:
```diff
@@ -1,24 +1,22 @@
-    <a class="cursor-pointer ml-3 truncate min-w-0" @click="play(song)">{{ song.title }}</a>
+    <a class="cursor-pointer truncate min-w-0" @click="play(song)">{{ song.title }}</a>
+    <span v-if="showSource" class="px-1 mx-4 text-gray-500 text-ms text-center border border-solid rounded border-gray-500">
+      {{ song.source }}
```
```suggestion
      {{ t(song.source) }}
```
Add translated source name

mikelxk commented on src/views/Settings.vue:
```diff
@@ -2,100 +2,60 @@
-        </div>
-        <SettingTitle :text="t('_BACKUP_PLAYLIST')" />
-        <div class="settings-content mx-7 mb-7 leading-8">
+      <div class="cover-container leading-normal flex content-left flex-col flex-wrap gap-4 2xl:h-[calc(100vh-10rem)]">
```
```suggestion
      <div class="cover-container leading-normal flex content-left flex-col flex-wrap gap-4 2xl:h-[calc(100vh-10rem)] 2xl:overflow-y-hidden">
```
I would recommend disable y scroll in 2xl.

Dumeng replied:
Didn't figure out any obvious effect. Can you explain more?

mikelxk commented:
Could you extend the view window for lyric
<img width="500" alt="image" src="https://user-images.githubusercontent.com/32220948/150624195-d67cd61f-b0ec-413e-9241-d6fe64eb4c93.png">
That's the current window size, it seems that I can only see 5 lines of clear text in my current setting, medium font weight and 20px.
<img width="500" alt="image" src="https://user-images.githubusercontent.com/32220948/150624242-855dc534-b0d8-468c-b80f-089cb6a2c3b0.png">
Also, the highlighted lyric is a little bit under the center of cover art, it would be very great if you could align that as well.

listen1 commented:
Sorry for merge delay. This pr is too big to be merged for one time, especially many modification made and features included. It's beyond my ability to ensure all the things go well after merge.

Please submit pr for one feature, or UI adjust, or code refactor. It's easier to test and make further commit.

Thank you for your understanding.

Dumeng commented:
I will try to do cherry-picks. But many reviews and modifications are made, I'm not sure this will work well.
